esphome/components/tuya/climate/__init__.py

```diff
@@ -67,30 +83,68 @@ def validate_temperature_multipliers(value):
     return value
 
 
-def validate_active_state_values(value):
-    if CONF_ACTIVE_STATE_DATAPOINT not in value:
-        if CONF_ACTIVE_STATE_COOLING_VALUE in value:
-            raise cv.Invalid(
-                f"{CONF_ACTIVE_STATE_DATAPOINT} required if using "
-                f"{CONF_ACTIVE_STATE_COOLING_VALUE}"
-            )
-    else:
-        if value[CONF_SUPPORTS_COOL] and CONF_ACTIVE_STATE_COOLING_VALUE not in value:
-            raise cv.Invalid(
-                f"{CONF_ACTIVE_STATE_COOLING_VALUE} required if using "
-                f"{CONF_ACTIVE_STATE_DATAPOINT} and device supports cooling"
-            )
+def validate_cooling_values(value):
+    if CONF_SUPPORTS_COOL in value:
+        cooling_supported = value[CONF_SUPPORTS_COOL]
+        if CONF_ACTIVE_STATE in value:
+            active_state_config = value[CONF_ACTIVE_STATE]
+            if CONF_COOLING_VALUE in active_state_config and not cooling_supported:
+                raise cv.Invalid(
+                    f"device does not support cooling but "
+                    f"{CONF_COOLING_VALUE} specified"
+                )
+            if cooling_supported:
+                if (
+                    CONF_COOLING_VALUE not in active_state_config
+                    and CONF_COOLING_STATE_PIN not in value
+                ):
+                    raise cv.Invalid(
+                        f"active_state {CONF_COOLING_VALUE} or "
+                        f"{CONF_COOLING_STATE_PIN} required if using "
+                        f"{CONF_STATE_DATAPOINT} and device supports cooling"
+                    )
     return value
 
 
-def validate_eco_values(value):
-    if CONF_ECO_TEMPERATURE in value and CONF_ECO_DATAPOINT not in value:
-        raise cv.Invalid(
-            f"{CONF_ECO_DATAPOINT} required if using {CONF_ECO_TEMPERATURE}"
-        )
-    return value
+ACTIVE_STATES = cv.Schema(
+    {
+        cv.Required(CONF_STATE_DATAPOINT): cv.uint8_t,
```

Comment:
Lets simplify these as just `datapoint` in config since they are already inside the block that describes what it is for.

Same for eco and fan

```suggestion
        cv.Required(CONF_DATAPOINT): cv.uint8_t,
```